docs/source/onnx.rst

```diff
@@ -0,0 +1,137 @@
+
+Key Features
+------------
+
+- **Seamless Model Chaining**: Combine multiple ONNX models into a single computational graph.
+- **Flexible Input/Output Mapping**: Control how the outputs of one model are passed as inputs to the next.
+- **Optimized Execution**: Automatically create optimized `ONNXRuntime` sessions to speed up inference.
+- **Export to ONNX**: Save the combined model into a single ONNX file for easy deployment and sharing.
+- **Execution Providers Support**: Utilize ONNXRuntime's execution providers (e.g., `CUDAExecutionProvider`, `CPUExecutionProvider`) for accelerated inference on different hardware.
+- **PyTorch-like Interface**: Use the `ONNXSequential` class like a PyTorch `nn.Sequential` model, including calling it directly for inference.
+
+Quickstart Guide
+----------------
+
+Here's how you can quickly get started with `ONNXSequential`:
+
+1. **Install ONNX and ONNXRuntime**
+
+   If you haven't already installed `onnx` and `onnxruntime`, you can install them using `pip`:
+
+   .. code-block:: bash
+
+      pip install onnx onnxruntime
+
+2. **Combining ONNX Models**
+
+   You can initialize the `ONNXSequential` with a list of ONNX models or file paths. Models will be automatically chained together and optimized for inference.
+
+   .. code-block:: python
+
+      import numpy as np
+      from kornia.onnx import ONNXSequential
+
+      # Initialize ONNXSequential with two models, loading from our only repo
+      onnx_seq = ONNXSequential(
+         "hf://operators/kornia.color.gray.RgbToGrayscale",
+         "hf://operators/kornia.geometry.transform.affwarp.Resize_512x512"
+      )
+
+      # Prepare some input data
+      input_data = np.random.randn(1, 3, 256, 512).astype(np.float32)
+
+      # Perform inference
+      outputs = onnx_seq(input_data)
+
+      # Print the model outputs
+      print(outputs)
+
+   .. note::
+      By default, we assume each ONNX model contains only one input node named "input" and one output node named "output". For complex models, you may need to pass an `io_maps` argument.
+
+3. **Input/Output Mapping Between Models**
+
+   When combining models, you can specify how the outputs of one model are mapped to the inputs of the next. This allows you to chain models in custom ways.
+
+   .. code-block:: python
+
+      io_map = [("model1_output_0", "model2_input_0"), ("model1_output_1", "model2_input_1")]
+      onnx_seq = ONNXSequential("model1.onnx", "model2.onnx", io_map=io_map)
+
+4. **Exporting the Combined Model**
+
+   You can easily export the combined model to an ONNX file:
+
+   .. code-block:: python
+
+      # Export the combined model to a file
+      onnx_seq.export("combined_model.onnx")
+
+5. **Optimizing with Execution Providers**
+
+   Leverage ONNXRuntime's execution providers for optimized inference. For example, to run the model on a GPU:
+
+   .. code-block:: python
+
+      # Initialize with CUDA execution provider
+      onnx_seq = ONNXSequential(
```

Comment:
One feature that might valuable is to provide per mapping a lamba function to inject adapters between ONNX ir, that can be written in ONNX itself as probably not always there will be compatibility between models